pint/compat.py

```diff
@@ -293,6 +298,26 @@ def is_duck_array_type(cls: type) -> bool:
     )
 
 
+def is_timedelta(obj: Any) -> bool:
+    """Check if the object is a datetime object."""
+    return isinstance(obj, datetime.timedelta) or isinstance(obj, np_timedelta64)
+
+
+def is_timedelta_array(obj: Any) -> bool:
+    """Check if the object is a datetime array."""
+    if isinstance(obj, ndarray) and obj.dtype == np_timedelta64:
+        return True
+
+
+def to_seconds(obj: Any) -> float:
+    """Convert a timedelta object to seconds."""
+    if isinstance(obj, datetime.timedelta):
+        return obj.total_seconds()
+    elif isinstance(obj, np_timedelta64) or obj.dtype == np_timedelta64:
+        return obj.astype(float)
```

Comment:
this is not going to work in general: `timedelta64` has its own units, which can range from days (or years?) to nanoseconds, and `astype(float)` is the equivalent of `magnitude`. I don't have any advice on how to extract that, though, other than parsing the string repr of the dtype.